yt/frontends/boxlib/fields.py

```diff
@@ -523,15 +525,24 @@ def setup_fluid_fields(self):
 
 
 def _nice_species_name(field):
-    spec_match = spec_finder.search(field)
-    nice_name = "".join(spec_match.groups())
+    try:
+        spec_match = spec_finder.search(field)
+        nice_name = "".join(spec_match.groups())
+
+    except AttributeError as e:
+        # Catch exception cases for e.g. molecules (H2O, C6H6)
+        print("Could not parse species ", field)
```

Comment:
We do not use `print` to log errors. You could use `ytLogger.error` here instead.

Reply:
It's not really an "error" and ytLogger prints a whole traceback, not sure that is the ideal solution

Reply:
> It's not really an "error"

please clarify. To me, seeing `return` inside an `except` blocks means we've encountered a recoverable exception (error).


> ytLogger prints a whole traceback

I'm not sure what you mean here. `ytLogger` is meant to print messages to the console and we generally don't include (even partial) tracebacks.

Reply:
Sorry I'm using the wrong terminology. I'm just trying to say that the "error" in this case is not a "big deal". The species is not recognized, so we won't try to produce a nice tex label, that's it.  ytLogger prints out very long error message and the whole call stack, which feels like overkill.

Here's what I get 
```
python $MAESTROEX_DIR/Util/yt/plotsinglevar.py plt_0000100 "X(H2O)"
--- Logging error ---
Traceback (most recent call last):
  File "/Users/simon/anaconda3/lib/python3.11/logging/__init__.py", line 1110, in emit
    msg = self.format(record)
          ^^^^^^^^^^^^^^^^^^^
  File "/Users/simon/anaconda3/lib/python3.11/logging/__init__.py", line 953, in format
    return fmt.format(record)
           ^^^^^^^^^^^^^^^^^^
  File "/Users/simon/anaconda3/lib/python3.11/logging/__init__.py", line 687, in format
    record.message = record.getMessage()
                     ^^^^^^^^^^^^^^^^^^^
  File "/Users/simon/anaconda3/lib/python3.11/logging/__init__.py", line 377, in getMessage
    msg = msg % self.args
          ~~~~^~~~~~~~~~~
TypeError: not all arguments converted during string formatting
Call stack:
  File "/Users/simon/Desktop/codes/AMREX-Astro/MAESTROeX/Util/yt/plotsinglevar.py", line 94, in <module>
    plot_single_var(args.plotfile, args.outfile, args.variables, use_log, args.norm, args.minimum, args.maximum)
  File "/Users/simon/Desktop/codes/AMREX-Astro/MAESTROeX/Util/yt/plotsinglevar.py", line 52, in plot_single_var
    plots = yt.SlicePlot(ds, norm_axis, var_names)
  File "/Users/simon/Desktop/codes/yt-git/yt/visualization/plot_window.py", line 1821, in __init__
    (bounds, center, display_center) = get_window_parameters(
  File "/Users/simon/Desktop/codes/yt-git/yt/visualization/plot_window.py", line 70, in get_window_parameters
    width = ds.coordinates.sanitize_width(axis, width, None)
  File "/Users/simon/Desktop/codes/yt-git/yt/geometry/coordinates/coordinate_handler.py", line 300, in sanitize_width
    self.ds.index
  File "/Users/simon/Desktop/codes/yt-git/yt/data_objects/static_output.py", line 613, in index
    self.create_field_info()
  File "/Users/simon/Desktop/codes/yt-git/yt/data_objects/static_output.py", line 665, in create_field_info
    self.field_info.setup_fluid_fields()
  File "/Users/simon/Desktop/codes/yt-git/yt/frontends/boxlib/fields.py", line 475, in setup_fluid_fields
    nice_name, tex_label = _nice_species_name(field)
  File "/Users/simon/Desktop/codes/yt-git/yt/frontends/boxlib/fields.py", line 534, in _nice_species_name
    ytLogger.error("Could not parse species ", field)
Message: 'Could not parse species '
Arguments: ('X(H2O)',)
```

Reply:
The traceback isn't a feature of the logger, but shows a mistake in your call to `ytLogger.error`:

Instead of
```python
ytLogger.error("Could not parse species ", field)
```
write
```python
ytLogger.error("Could not parse species %s", field)
```

Reply:
Oh, thank you!